plan/plan.go

```diff
@@ -175,3 +190,16 @@ func shouldUpdateTTL(desired, current *endpoint.Endpoint) bool {
 	}
 	return desired.RecordTTL != current.RecordTTL
 }
+
+func stripLowerName(dnsName string) (string, error) {
```

Comment:
Can you add a comment for this function? I would also propose renaming it as the name suggests that it strips the lower part of the name..